### PR TITLE
Read,list,delete query template

### PR DESCRIPTION
**Pull request by yashashkumar**
…ion and route

### CODE REVIEW

**Review by HarishGangula — CHANGES_REQUESTED** [review]
List API changes need to be done

File: api-service/src/controllers/ReadQueryTemplate/ReadTemplateController.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    const template_id = _.get(req, 'params.templateId');
+    try {
+        const resmsgid = _.get(res, "resmsgid");
+
+        const template = await getQueryTemplate(template_id);
+        if (template === null) {
+            logger.error({ apiId, resmsgid, template_id, message: `Template ${template_id} does not exists`, code: "QUERY_TEMPLATE_NOT_EXISTS" })
+            return ResponseHandler.errorResponse({ message: `Template ${template_id} does not exists`, statusCode: 404, errCode: "NOT_FOUND", code: "QUERY_TEMPLATE_NOT_EXISTS" }, req, res);
+        }
+        logger.info({ apiId, resmsgid, template_id, message: `Template read successfully with id: ${template_id}` })
```
Comment (HarishGangula) [inline api-service/src/controllers/ReadQueryTemplate/ReadTemplateController.ts:18]: you should log the response also

File: api-service/src/controllers/ListQueryTemplates/ListTemplateValidationSchema.json [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                            }
+                        },
+                        "additionalProperties": false,
+                        "required": [
+                            "field",
+                            "order"
+                        ]
+                    }
+                },
+                "limit": {
```
Comment (HarishGangula) [inline api-service/src/controllers/ListQueryTemplates/ListTemplateValidationSchema.json:71]: where is the min and max value for limit and offset

File: api-service/src/controllers/ListQueryTemplates/ListTemplatesController.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        ResponseHandler.errorResponse({ code: "QUERY_TEMPLATE_LIST_FAILED", message: "Failed to list query templates" }, req, res);
+    }
+}
+
+const getTemplateList = async (req: Record<string, any>) => {
+    const limit: any = _.get(req, "limit");
+    const offset: any = _.get(req, "offset");
+    const { filters = {}, sortBy = [] } = req || {};
+    const templates = await QueryTemplate.findAll({ limit: limit || 100, offset: offset || 0, ...(filters && { where: filters }) })
+    if (sortBy) {
```
Comment (HarishGangula) [inline api-service/src/controllers/ListQueryTemplates/ListTemplatesController.ts:40]: sort should be part of query not script

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: api-service/src/controllers/ListQueryTemplates/ListTemplateValidationSchema.json (hunk 2 above)
Comment (yashashkumar) [reply api-service/src/controllers/ListQueryTemplates/ListTemplateValidationSchema.json:71]: done

File: api-service/src/controllers/ReadQueryTemplate/ReadTemplateController.ts (hunk 1 above)
Comment (yashashkumar) [reply api-service/src/controllers/ReadQueryTemplate/ReadTemplateController.ts:18]: done

File: api-service/src/controllers/ListQueryTemplates/ListTemplatesController.ts (hunk 3 above)
Comment (yashashkumar) [reply api-service/src/controllers/ListQueryTemplates/ListTemplatesController.ts]: done

File: api-service/src/controllers/ListQueryTemplates/ListTemplateValidationSchema.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                    "type": "number",
+                    "minimum": 5,
+                    "maximum": 500
+                },
+                "offset": {
+                    "type": "number",
+                    "minimum": 0,
+                    "maximum": 10
```
Comment (HarishGangula) [inline api-service/src/controllers/ListQueryTemplates/ListTemplateValidationSchema.json:76]: offset may not have maximum

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: api-service/src/controllers/ListQueryTemplates/ListTemplateValidationSchema.json (hunk 4 above)
Comment (yashashkumar) [reply api-service/src/controllers/ListQueryTemplates/ListTemplateValidationSchema.json:76]: done

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**